File: 2017-03-15/W2D3-TabViewController/W2D3-TabViewController/FirstViewController.m
//
//  FirstViewController.m
//  W2D3-TabViewController
//
//  Created by Jason Liang on 3/15/17.
//  Copyright © 2017 Jason Liang. All rights reserved.
//

#import "FirstViewController.h"
#import "ChildViewController.h"
@interface FirstViewController ()

@end

@implementation FirstViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  // Do any additional setup after loading the view, typically from a nib.
  NSLog(@"ViewDidLoad");
  
  // 75%
  [self embedChildViewController];
}


- (void)embedChildViewController {
  ChildViewController *childViewController = [[ChildViewController alloc] init];
  // add the childViewController view
  [self addChildViewController:childViewController];
  childViewController.view.frame = CGRectMake(100, 100, 100, 100);
  [self.view addSubview:childViewController.view];
  
}

- (void)removeChildViewController:(ChildViewController *)childVC {
  [childVC.view removeFromSuperview];
  [childVC removeFromParentViewController];
}


- (void)didReceiveMemoryWarning {
  [super didReceiveMemoryWarning];
  // Dispose of any resources that can be recreated.
}


//- (void)viewWillAppear:(BOOL)animated {
//  [super viewWillAppear:animated];
//  NSLog(@"1 - viewWillAppear");
//  
//}
//- (void)viewDidAppear:(BOOL)animated {
//  [super viewDidAppear:animated];
//  NSLog(@"2 - viewDidAppear");
//  // 20%
//}
//- (void)viewWillDisappear:(BOOL)animated {
//  [super viewWillDisappear:animated];
//  NSLog(@"3 - viewWillDisappear");
//}
//- (void)viewDidDisappear:(BOOL)animated {
//  [super viewDidDisappear:animated];
//  NSLog(@"4 - viewDidDisappear");
//}


@end

File: 2018-06-28/W4D4/W4D4/FirstViewController.swift
//
//  FirstViewController.swift
//  W4D4
//
//  Created by Jason Liang on 6/28/18.
//  Copyright © 2018 jasonliang. All rights reserved.
//

import UIKit
import MaterialComponents.MaterialButtons

class FirstViewController: UIViewController {
  
  let floatingButton = MDCButton()
  
  override func viewDidLoad() {
    super.viewDidLoad()
    floatingButton.frame = CGRect(x: 100, y: 100, width: 200, height: 60);
    floatingButton.setElevation(.dialog, for: .normal)
    floatingButton.setElevation(.cardPickedUp, for: .highlighted)
    view.addSubview(floatingButton)
  }

}

